### PR TITLE
Added organisational unit administration

### PR DESCRIPTION
**Pull request by Coenego**

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @simong for review

File: shared/gh/partials/tenant-admin/admin-manage-orgunits.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<div id="gh-manage-orgunits">
+
+    <div id="gh-manage-orgunits-toolbar">
+        <h2 class="pull-left">Organisational unit administration</h2>
```
Comment (simong) [inline shared/gh/partials/tenant-admin/admin-manage-orgunits.html]: Maybe call this title `Course, subject and part administration`

**Comment by simong** [comment]
When you:
 * click jeditable A (don't change anything)
 * click jeditable B (don't change anything)
 * click jeditable A again

you get a notification saying "Organisational unit successfully updated"

**Comment by simong** [comment]
When you update a part the notification should say "Part updated", similarly for subjects and courses.

Ideally, we don't expose the "organisational unit" terminology anywhere

**Comment by simong** [comment]
When you create a part under a subject we can just hide the select box as it only contains the part value anyway

**Comment by Coenego** [comment]
Follow-up done, back to @simong for review

File: shared/gh/js/views/tenant-admin/gh.admin-manage-orgunits.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+     */
+    var addOrgUnit = function() {
+        // Get the form data
+        var displayName = $('#gh-manage-orgunits-modal-text').val();
+        var type = $('#gh-manage-orgunits-modal-type').val();
+        var parentId = parseInt($('#gh-manage-orgunits-add-modal').attr('data-id'), 10) || null;
+        var groupId  = parseInt($('#gh-manage-orgunits-add-modal').attr('data-groupid'), 10) || null;
+
+        // Create the organisational unit
+        gh.api.orgunitAPI.createOrgUnit(gh.data.me.AppId, displayName, type, parentId, groupId, null, null, false, function(err, data) {
```
Comment (simong) [inline shared/gh/js/views/tenant-admin/gh.admin-manage-orgunits.js]: I don't think this is right. Each course, subject and part should have a different group. IIRC we need to be able to assign different users to different parts. Take for example:
```
Asian and Middle eastern studies
  + Chinese
      + Part IA   (Dr Miranda Foo)
      + Part IB   (Dr Miranda Foo)
      + Part II   (Dr Jack Bower)
      + Part III   (Dr Jack Bower)
   Japanese
      + Part IA   (Dr Miranda Foo)
      + Part IB   (Dr Miranda Foo)
      + Part II   (Dr Jack Bower)
      + Part III   (Dr Jack Bower)
```

If the part always inherits the subject/course group we won't be able to make these distinctions

Comment (simong) [inline shared/gh/js/views/tenant-admin/gh.admin-manage-orgunits.js]: All this to say that you can leave groupId `null` (but please do verify this with a few users, it's been a while)

File: shared/gh/partials/tenant-admin/admin-manage-orgunits-add-modal.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">×</span>
+                    <span class="sr-only">Close</span>
+                </button>
+                <%
+                    var title = 'course';
+                    if (data.data.orgUnitType) {
```
Comment (simong) [inline shared/gh/partials/tenant-admin/admin-manage-orgunits-add-modal.html]: Is there a way to make this just `data.orgUnitType` ? Don't bother if it's too much work

**Comment by simong** [comment]
Good stuff, this'll really help helpdesk. 1 thing remaining really

**Comment by simong** [comment]
One smallish thing: If you rename the course `aaa` to `Yeah` it will be sorted to the bottom. Maybe the scroll should be adjusted as well?

**Comment by Coenego** [comment]
Follow-up done. Back to @simong

**Comment by simong** [comment]
LGTM, I'll be merging this all into our grasshopper-ui repository later

**Pull request merged**